### PR TITLE
support extended resource for kube-batch

### PR DESCRIPTION
**Pull request by wackxu**
@k82cn 

**Release note**:

```release-note
support extended resource for kube-batch
```

### CODE REVIEW

**Comment by TommyLike** [comment]
This patch needs a rebase since the change of repo name

File: pkg/scheduler/api/resource_info.go [hunk 1]
```diff
@@ -88,18 +111,27 @@ func (r *Resource) IsZero(rn v1.ResourceName) bool {
...
 		return r.Memory < minMemory
-	case GPUResourceName:
-		return r.MilliGPU < minMilliGPU
 	default:
-		panic("unknown resource")
+		if r.ScalarResources == nil {
+			return true
+		}
+
+		return r.ScalarResources[rn] < minMilliScalarResources
```
Comment (k82cn) [inline pkg/scheduler/api/resource_info.go:123]: check whether `rn` is valid.

**Comment by k82cn** [comment]
please add some test for that.

File: pkg/scheduler/plugins/proportion/proportion.go [hunk 2]
```diff
@@ -232,7 +232,7 @@ func (pp *proportionPlugin) updateShare(attr *queueAttr) {
 	res := float64(0)
 
 	// TODO(k82cn): how to handle fragment issues?
-	for _, rn := range api.ResourceNames() {
+	for _, rn := range attr.allocated.ResourceNames() {
```
Comment (hex108) [inline pkg/scheduler/plugins/proportion/proportion.go]: `attr.allocated`'s extended resource is initialed to nil, so its `ResourceNames()`  will not contain extended resource?

File: pkg/scheduler/plugins/drf/drf.go [hunk 3]
```diff
@@ -160,7 +160,7 @@ func (drf *drfPlugin) updateShare(attr *drfAttr) {
 
 func (drf *drfPlugin) calculateShare(allocated, totalResource *api.Resource) float64 {
 	res := float64(0)
-	for _, rn := range api.ResourceNames() {
+	for _, rn := range allocated.ResourceNames() {
```
Comment (hex108) [inline pkg/scheduler/plugins/drf/drf.go]: Might it be better to use `totalResource` because `allocated`'s `ScalarResources` perhaps is nil?

Comment (wackxu) [reply pkg/scheduler/plugins/drf/drf.go]: Thanks for your review, updated

File: pkg/scheduler/plugins/proportion/proportion.go (hunk 2 above)
Comment (wackxu) [reply pkg/scheduler/plugins/proportion/proportion.go]: update to use attr.deserved

File: pkg/scheduler/api/resource_info.go (hunk 1 above)
Comment (wackxu) [reply pkg/scheduler/api/resource_info.go:123]: Done

**Comment by wackxu** [comment]
Add some UT, @k82cn @hex108 @TommyLike PTAL

**Comment by hex108** [comment]
/lgtm

Thanks!

**Pull request merged**